app/move/controllers/unassign.js

```diff
@@ -0,0 +1,60 @@
+const FormWizardController = require('../../../common/controllers/form-wizard')
+const allocationService = require('../../../common/services/allocation')
+const moveService = require('../../../common/services/move')
+
+class UnassignController extends FormWizardController {
+  middlewareChecks() {
+    this.use(this.checkAllocation)
+    super.middlewareChecks()
+  }
+
+  async checkAllocation(req, res, next) {
+    const { move } = res.locals
+
+    if (!move.allocation) {
+      return res.redirect(`/move/${move.id}`)
+    }
+    if (!move.person) {
+      return res.redirect(`/allocation/${move.allocation.id}`)
+    }
+
+    next()
+  }
+
+  middlewareLocals() {
+    super.middlewareLocals()
+    this.use(this.setMoveRelationships)
+  }
+
+  async setMoveRelationships(req, res, next) {
+    const { move } = res.locals
+
+    const allocation = await allocationService.getById(move.allocation.id)
+    res.locals.allocation = allocation
+
+    res.locals.person = move.person
+
+    next()
+  }
+
+  async saveValues(req, res, next) {
+    try {
+      const { id } = res.locals.move
+      await moveService.unassign(id)
+      super.saveValues(req, res, next)
+    } catch (err) {
+      next(err)
+    }
+  }
+
+  successHandler(req, res) {
+    const { id: allocationId } = res.locals.allocation
```

Comment:
Again, think about the fact an allocation actually exists on the move this could be:

```javascript
const { id: allocationId } = res.locals.move.allocation
```

Then does that mean you don't need to set `res.locals.allocation` in `setMoveRelationships`?

Reply:
Have updated to get the id from move's allocation rather than, er, the allocation

See above why we do need the actual allocation

Reply:
^^ See above for why I'm wrong